runtime/src/test/java/com/flipkart/flux/resource/StateMachineResourceTest.java

```diff
@@ -208,6 +213,63 @@ public void testPostEvent_againstNonExistingCorrelationId() throws Exception {
     }
 
     @Test
+    public void testPostScheduledEvent_withoutCorrelationIdTag() throws Exception {
+        String stateMachineDefinitionJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("state_machine_definition.json"));
+        Unirest.post(STATE_MACHINE_RESOURCE_URL).header("Content-Type","application/json").body(stateMachineDefinitionJson).asString();
+
+        String eventJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("event_data.json"));
+
+        //request with searchField param missing
+        final HttpResponse<String> eventPostResponse = Unirest.post(STATE_MACHINE_RESOURCE_URL+SLASH+"magic_number_1"+"/context/events?triggerTime=123").header("Content-Type","application/json").body(eventJson).asString();
+        assertThat(eventPostResponse.getStatus()).isEqualTo(Response.Status.BAD_REQUEST.getStatusCode());
+
+        //request with searchField param having value other than correlationId
+        final HttpResponse<String> eventPostResponseWithWrongTag = Unirest.post(STATE_MACHINE_RESOURCE_URL+SLASH+"magic_number_1"+"/context/events?searchField=dummy&triggerTime=123").header("Content-Type","application/json").body(eventJson).asString();
+        assertThat(eventPostResponseWithWrongTag.getStatus()).isEqualTo(Response.Status.BAD_REQUEST.getStatusCode());
+    }
+
+    @Test
+    public void testPostScheduledEvent_withCorrelationId() throws Exception {
+        //create state machine
+        String stateMachineDefinitionJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("state_machine_definition.json"));
```

Comment:
As we perform different operations in each test case (submitting different events for ex.), it is not possible to have a single state machine for all.